src/connect.rs

```diff
@@ -687,7 +691,16 @@ mod native_tls_conn {
 
     impl<T: Connection + AsyncRead + AsyncWrite + Unpin> Connection for NativeTlsConn<T> {
         fn connected(&self) -> Connected {
-            self.inner.get_ref().get_ref().get_ref().connected()
+            if self.inner.get_ref().negotiated_alpn().ok() == Some(Some(b"h2".to_vec())) {
```

Comment:
This will allocate a small vector before doing the comparison, we could just compare the slices with some more pattern matching.